auslib/test/admin/views/test_rules.py

```diff
@@ -141,9 +141,22 @@ def testNewRuleWithoutPermission(self):
     def testMissingFields(self):
         # But we still need to pass product, because permission checking
         # is done before what we're testing
-        ret = self._post('/rules', data=dict({'product': 'a'}))
+        post_data = {'product': 'a'}
+        ret = self._post('/rules', data=post_data)
+        self.assertEquals(ret.status_code, 400, "Status Code: %d, Data: %s" % (ret.status_code, ret.data))
+        # Connexion and manual request.json validation methods will preempt and throw 400 error as
+        # soon as the first field invalidates. Only one field may be present in the response.data depending
+        # upon the order of fields sorted in swagger yaml file and in views.
+        self.assertTrue('title' in ret.data, msg=ret.data)
```

Comment:
One approach can be to use custom validator for requests to customize the responses generated when error occurs. Inheriting [this](https://github.com/zalando/connexion/blob/master/connexion/decorators/validation.py#L156) validator method and re-writing it like ex-Dev did for re-writing and one of his parameter validators according to his use-case. 

Also, reason why connexion throws exception as soon as it encounters error in validation, in contrast to wtfForms validation which runs the validation on all fields and returns a dictionary of field key names and their exceptions is because when it invokes jsonschema's validate() method (which is currently based on json schema Draft4Validator) , it in turns throws exception as soon as it encounters. [Link](https://github.com/Julian/jsonschema/blob/master/jsonschema/validators.py#L128).  This implies that we can only highlight or throw one field error at a time from now in our  UI forms .